### PR TITLE
Fixed #24982 -- Split staticfiles tests into multiple files

### PR DESCRIPTION
**Pull request by MoritzS**
Ticket: https://code.djangoproject.com/ticket/24982

### CODE REVIEW

File: tests/staticfiles_tests/test_storage.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        self.assertEqual(cached_name, None)
+        self.assertEqual(self.hashed_file_path(name), hashed_name)
+        cached_name = storage.staticfiles_storage.hashed_files.get(cache_key)
+        self.assertEqual(cached_name, hashed_name)
+
+    def test_cache_key_memcache_validation(self):
+        """
+        Handle cache key creation correctly, see #17861.
+        """
+        name = "/some crazy/long filename/ with spaces Here and ?#%#$/other/stuff/some crazy/long filename/ with spaces Here and ?#%#$/other/stuff/some crazy/long filename/ with spaces Here and ?#%#$/other/stuff/some crazy/long filename/ with spaces Here and ?#%#$/other/stuff/some crazy/long filename/ with spaces Here and ?#%#$/other/stuff/some crazy/" + "\x16" + "\xb4"
```
Comment (timgraham) [inline tests/staticfiles_tests/test_storage.py]: could you split this string over multiple lines?
```
name = (
    "...."
    "...."
)
```

File: tests/staticfiles_tests/cases.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        fails. Regression test for #18986.
+        """
+        finders.get_finder.cache_clear()
+        err = six.StringIO()
+        with self.assertRaises(Exception):
+            call_command('collectstatic', interactive=False, verbosity=0, stderr=err)
+        self.assertEqual("Post-processing 'faulty.css' failed!\n\n", err.getvalue())
+
+
+class FinderTestCase(object):
```
Comment (timgraham) [inline tests/staticfiles_tests/cases.py]: if a case is only used in 1 file like `FinderTestCase`,  I'd put it there.

**Comment by timgraham** [comment]
overall organization looks good, thanks.

File: tests/staticfiles_tests/cases.py (hunk 2 above)
Comment (MoritzS) [inline tests/staticfiles_tests/cases.py]: Should I do the same with `TestHashedFiles` (only used in `test_storage.py`) and `TestNoFilesCreated` (only used in `test_management.py`)?

Comment (timgraham) [inline tests/staticfiles_tests/cases.py]: Yes please, I didn't audit for all instances.

**Comment by timgraham** [comment]
merged in b35b43dff81d46e930ffa1e05eb50968b8557102.

**Pull request closed without merging**